starknet_py/tests/e2e/docs/guide/test_serializing.py

```diff
@@ -20,64 +19,72 @@ def test_short_strings():
 
 
 def test_abi_parsing():
-    raw_abi_string = read_contract(
-        "erc20_abi.json", directory=CONTRACTS_COMPILED_V0_DIR
-    )
+    raw_abi_string = json.loads(
+        load_contract(contract_name="TestContract", version=ContractVersion.V2)[
+            "sierra"
+        ]
+    )["abi"]
     # docs-serializer: start
-    from starknet_py.abi.v0 import AbiParser
+    from starknet_py.abi.v2 import AbiParser
 
     """
-    @external
-    func transferFrom{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-        sender: felt, recipient: felt, amount: felt
-    ) -> (success: felt) {
-        alloc_locals;
-        local success: felt = 1;
-        return (success,);
+    #[external(v0)]
+    fn test(ref self: ContractState, ref arg: felt252, arg1: felt252, arg2: felt252) -> felt252 {
+        let mut x = self.my_storage_var.read();
+        x += 1;
+        self.my_storage_var.write(x);
+        x + internal_func()
```

Comment:
Correct me if I'm wrong but shouldn't we use `transfer_from` example of v2 contract? Also, changing this test name from `testTransfer` to `test` seems a but unclear to me.

Reply:
Actually, the contract type does not matter in this example, the idea of function serialization is preserved, there are many arguments that can be explicitly taken by name. The function was taken from real contract [test_contract.cairo](https://github.com/software-mansion/starknet.py/blob/development/starknet_py/tests/e2e/mock/contracts_v2/src/test_contract.cairo#L25), so that's why fn is called `test`